src/tools/schema.js

```diff
@@ -13,43 +35,50 @@ const convertResourceDos = appRaw => {
     searchOrCreates = {};
 
   _.each(appRaw.resources, resource => {
-    let search, create, trigger;
-
-    if (resource.hook && resource.hook.operation) {
-      trigger = dataTools.deepCopy(resource.hook);
+    walkMethods(resource, resource.hook).forEach(trigger => {
       trigger.key = `${resource.key}Hook`;
       trigger.noun = resource.noun;
       trigger.operation.resource = resource.key;
       trigger.operation.type = 'hook';
       triggers[trigger.key] = trigger;
-    }
+    });
 
-    if (resource.list && resource.list.operation) {
-      trigger = dataTools.deepCopy(resource.list);
+    walkMethods(resource, resource.list).forEach(trigger => {
       trigger.key = `${resource.key}List`;
       trigger.noun = resource.noun;
       trigger.operation.resource = resource.key;
       trigger.operation.type = 'polling';
       triggers[trigger.key] = trigger;
-    }
+    });
 
-    if (resource.search && resource.search.operation) {
-      search = dataTools.deepCopy(resource.search);
+    walkMethods(resource, resource.search).forEach(search => {
       search.key = `${resource.key}Search`;
       search.noun = resource.noun;
       search.operation.resource = resource.key;
       searches[search.key] = search;
-    }
+    });
 
-    if (resource.create && resource.create.operation) {
-      create = dataTools.deepCopy(resource.create);
+    walkMethods(resource, resource.create).forEach(create => {
       create.key = `${resource.key}Create`;
       create.noun = resource.noun;
       create.operation.resource = resource.key;
       creates[create.key] = create;
-    }
-
-    if (search && create) {
+    });
+
+    walkMethods(resource, resource.misc).forEach(misc => {
+      misc.key = `${resource.key}Misc`;
+      misc.noun = resource.noun;
+      misc.operation.resource = resource.key;
+      creates[misc.key] = misc;
+    });
+
+    // TODO: ergh... this won't work with multiple searches/creates
```

Comment:
I think we could enforce some sort of naming pattern with the keys to automatically tie them together. It would be a breaking change, but not too bad of one. In that case, we'd iterate the searches and see if there's a create with a matching key. totally doable!

Reply:
Could tie them together if one search and one create, and but if more than 1 for either, fall back to specific naming patterns.